core/model/modx/sources/modfilemediasource.class.php

```diff
@@ -494,6 +494,21 @@ public function renameObject($oldPath,$newName) {
             return false;
         }
 
+        // Check that the new file has a file type that is allowed
+        $allowedFileTypes = explode(',',$this->xpdo->getOption('upload_files',null,''));
+        $allowedFileTypes = array_merge(explode(',',$this->xpdo->getOption('upload_images')),explode(',',$this->xpdo->getOption('upload_media')),explode(',',$this->xpdo->getOption('upload_flash')),$allowedFileTypes);
+        $allowedFileTypes = array_unique($allowedFileTypes);
+
+        $file_split = explode('.', $newName);
+        $ext = $file_split[count($file_split) - 1];
```

Comment:
A lot has happened for PHP over the last 15 years, so if you don't mind I think pathinfo would be cleaner ;) That link you just posted does flag a potential issue in this PR which is the extensions are being checked in a case sensitive way, so perhaps (untested) it could use some case normalisation too.